crates/transpiler/src/passes/litinski_transformation.rs

```diff
@@ -0,0 +1,218 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+
+use qiskit_circuit::dag_circuit::{DAGCircuit, NodeType};
+use qiskit_circuit::imports::PAULI_EVOLUTION_GATE;
+use qiskit_circuit::operations::{
+    multiply_param, Operation, OperationRef, Param, PyGate, StandardGate,
+};
+use qiskit_circuit::packed_instruction::PackedInstruction;
+use qiskit_circuit::{Clbit, Qubit, VarsMode};
+
+use qiskit_quantum_info::clifford::Clifford;
+use qiskit_quantum_info::sparse_observable::PySparseObservable;
+
+use smallvec::smallvec;
+use std::f64::consts::PI;
+
+use crate::TranspilerError;
+
+// List of gate names supported by the pass: the pass is skipped if the circuit
+// contains gate names outside of this list.
+const SUPPORTED_GATE_NAMES: &[&str; 19] = &[
```

Comment:
I don't like the fact that we have hard-coded lists of gates in several places of the code.
Is there a list of Clifford gates in rust? Perhaps we should add it as part of `clifford.rs`.
I would prefer to write something like:
`SUPPORTED_GATE_NAMES = CLIFFORD_GATE_NAMES + ROTATION_GATE_NAMES`

Reply:
this may be a question to a follow-up PR:
- do we want to support `p` (which is equivalent to `rz` up to a global phase)?
- do we want to support `rx` and `ry` that are equivalent to `rz` up to clifford gates?

Reply:
I am hesitant about the other suggestion: we should be able to add new clifford gates to `clifford.rs` without immediately thinking how to adjust Litinski transformation to deal with them (or do you think we should)? So I prefer to keep the list of all Clifford gates separate from the list of gates supported by the pass.